state-and-event-handling/updating-state.md

```diff
@@ -41,7 +41,19 @@ Recall that pieces of state are responsible for managing how one piece of the co
 
 ## Working With State and Re-Rendering
 
-Re-rendering a component is hard work! Firstly, setting state is _asynchronous_. Secondly, to repeat, _every time that state updates, the component re-renders_. These two facts will aid us when debugging state.
+Wrapping our heads around state and rendering can be challenging!
+
+<br />
+
+Firstly, setting state is _asynchronous_. Not asynchronous in exactly the same way as making `axios` calls, but asynchronous in the sense that setting the value won't be visible to our logic until after React re-renders. For example, logging the value of a piece of state variable immediately after calling its set function will look like nothing has changed, because it hasn't. And it won't until React re-renders.
```

Comment:
I may be underestimating folks familiarity and comfort with React at this point of the unit, but I'm wondering if this point will click in their minds. Do you think linking to the docs (https://react.dev/reference/react/useState#ive-updated-the-state-but-logging-gives-me-the-old-value) would be helpful?

Or is this introducing too much at once?

Reply:
I'm pretty sure that folks are not comfortable with why this is, so having a link to some extra background sounds like a good idea.